source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties

```diff
@@ -1,4 +1,24 @@
 AdvisoryFileLock.InterruptedAcquiringLockFormat=Interrupted while acquiring blocking lock on FileChannel for file {0}
+EmailAddressParser.AddressTooLong=The e-mail address cannot be longer that 256 characters.
+EmailAddressParser.DomainCannotEndWithDot=The domain part cannot end with the dot (''.'') character
+EmailAddressParser.DomainCannotStartWithDot=The domain part cannot start with the dot (''.'') character
+EmailAddressParser.DomainTooLong=The domain part cannot be longer than 253 characters.
+EmailAddressParser.DotAtBeginning=The dot (''.'') character is not allowed at the beginning of the local part
+EmailAddressParser.DotAtEnd=The dot (''.'') character is not allowed at the end of the local part
+EmailAddressParser.EmptyLocalpart=The local part cannot be empty.
+EmailAddressParser.LocalPartTooLong=The local part cannot be longer that 64 characters.
+EmailAddressParser.ManyDots=Multiple dot (''.'') characters are not allowed
+EmailAddressParser.MissingDomainpart=Missing domain part.
+EmailAddressParser.SubdomainCannotBeEmpty=The subdomain name cannot be empty.
+EmailAddressParser.SubdomainCannotEndWithHyphen=The subdomain name cannot end with the hyphen (''-'') character.
+EmailAddressParser.SubdomainCannotStartWithHyphen=The subdomain name cannot start with the hyphen (''-'') character.
+EmailAddressParser.SubdomainTooLong=The subdomain name cannot be longer than 63 characters.
+EmailAddressParser.TooManyDomainLevels=The domain part cannot more that 127 levels.
+EmailAddressParser.UnallowedCharacterFormat=Unallowed character ''{0}'' at position {1}.
```

Comment:
Inconsistent wording in error messages.  Sometimes the subject is first (i.e. "The subdomain name..") and others, like this message, start with the quality of what is wrong (i.e. "Unallowed", "Unexpected").